test/e2e/framework/autoscaling/autoscaling_utils.go

```diff
@@ -573,23 +613,31 @@ func runServiceAndWorkloadForResourceConsumer(c clientset.Interface, ns, name st
 		AdditionalContainers: additionalContainers,
 	}
 
+	dpConfig := testutils.DeploymentConfig{
+		RCConfig: rcConfig,
+	}
+	dpConfig.NodeDumpFunc = framework.DumpNodeDebugInfo
+	dpConfig.ContainerDumpFunc = e2ekubectl.LogFailedContainers
+
 	switch kind {
 	case KindRC:
 		framework.ExpectNoError(e2erc.RunRC(rcConfig))
 	case KindDeployment:
-		dpConfig := testutils.DeploymentConfig{
-			RCConfig: rcConfig,
-		}
-		ginkgo.By(fmt.Sprintf("creating deployment %s in namespace %s", dpConfig.Name, dpConfig.Namespace))
-		dpConfig.NodeDumpFunc = framework.DumpNodeDebugInfo
-		dpConfig.ContainerDumpFunc = e2ekubectl.LogFailedContainers
+		ginkgo.By(fmt.Sprintf("Creating deployment %s in namespace %s", dpConfig.Name, dpConfig.Namespace))
 		framework.ExpectNoError(testutils.RunDeployment(dpConfig))
 	case KindReplicaSet:
 		rsConfig := testutils.ReplicaSetConfig{
 			RCConfig: rcConfig,
 		}
-		ginkgo.By(fmt.Sprintf("creating replicaset %s in namespace %s", rsConfig.Name, rsConfig.Namespace))
+		ginkgo.By(fmt.Sprintf("Creating replicaset %s in namespace %s", rsConfig.Name, rsConfig.Namespace))
 		framework.ExpectNoError(runReplicaSet(rsConfig))
+	case KindCRD:
+		crd := CreateCustomResourceDefinition(apiExtensionClient)
+		_, err := CreateCustomSubresourceInstance(ns, name, resourceClient, crd)
+		framework.ExpectNoError(err)
+
+		ginkgo.By(fmt.Sprintf("Creating deployment %s backing CRD in namespace %s", dpConfig.Name, dpConfig.Namespace))
+		framework.ExpectNoError(testutils.RunDeployment(dpConfig))
```

Comment:
if you want to keep cleanup to a single deletion, you can set the deployment's ownerReferences to the CRD, so when the CRD gets deleted, so does the deployment

as long as you are creating the deployment in the test's namespace, it's not really a big deal since the namespace will get deleted when the test is done as well